src/ray/gcs/gcs_server/gcs_actor_scheduler.cc

```diff
@@ -563,5 +623,48 @@ void RayletBasedActorScheduler::HandleWorkerLeaseReply(
   }
 }
 
+void GcsActorScheduler::HandleWorkerLeaseRejectedReply(
+    std::shared_ptr<GcsActor> actor, const rpc::RequestWorkerLeaseReply &reply) {
+  // The request was rejected because of insufficient resources.
+  if (actor->GetScheduledByGcs()) {
```

Comment:
maybe add some comments says the following block returns the resources back to cluster_resource_manager, and why we need to do it.